c/models/py_ftrl.cc

```diff
@@ -149,13 +152,48 @@ void Ftrl::m__dealloc__() {
 
 
 /*
-* .fit(...)
-* Do dataset validation and a call to `dtft->dispatch_fit(...)` method.
+*  Check if provided interactions are consistent with the column names
+*  of the training frame.
 */
-static PKArgs args_fit(2, 3, 0, false, false, {"X_train", "y_train",
-                       "X_validate", "y_validate", "nepochs_validate"},
+std::vector<sizetvec> Ftrl::convert_interactions() {
+  std::vector<sizetvec> interactions;
+  auto py_iter = py_interactions.to_oiter();
+  interactions.reserve(py_iter.size());
+
+  for (auto py_interaction : py_iter) {
+    size_t nfeatures = py_interaction.to_pylist().size();
+    sizetvec interaction;
+    interaction.reserve(nfeatures);
+    for (auto py_feature : py_interaction.to_oiter()) {
+      std::string feature = py_feature.to_string();
+
+      auto it = find(colnames.begin(), colnames.end(), feature);
```

Comment:
You could create a blank `DataTable` object with `colnames` as names, and then use `DataTable::xcolindex()` to convert each column name into its index. This would use a dictionary in order to do value look up, so would be slightly faster, and also emit a nice error message checking for possible misspellings of a column name (e.g. "column `c0` not found, did you mean `C0`?")

Reply:
That's a good idea, however, what about the custom error messages in this case? For example, now I emit

```
        throw ValueError() << "Feature " << py_feature << " is used for "
                              "interactions, however, it is missing in the "
                              "training frame";
```

If I use `DataTable`, it will just emit something general?